utilities/general-helpers.metta

```diff
@@ -324,10 +326,10 @@
 
 (= (sort $list $len) (sort $list $len <=)) 
 (= (sort $list $len $key)
-   (if (or (isUnit $list) (hasOneAtom $list))
+   (if (or (isUnit $list) (<= $len 1))
```

Comment:
The `isUnit` logic could also be generalized with the new logic you introduced so no need for it.